<html>
<head>
<meta http-equiv="content-type" content="text/html; charset=iso-8859-1"/>
<meta http-equiv="Cache-Control" content="max-age=3600"/>
<meta name="description" content="description"/>
<meta name="keywords" content="keywords"/> 
<meta name="author" content="author"/> 
<link rel="stylesheet" type="text/css" href="../style/default.css" media="screen"/>
<title>QTLtools</title>
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-80577718-1', 'auto');
  ga('send', 'pageview');

</script>
</head>

<body>
<div class="content">
	<div class="item">
		<h1>General information</h1>
		<p>
		<b>QTLtools</b> is a tool set for molecular QTL discovery and analysis. It allows to go from the raw sequence data to collection of molecular Quantitative Trait Loci (QTLs) in few easy-to-perform steps. For the moment, QTLtools can perform the following tasks:
		<ul>
			<li><b>bamstat</b>; to check the quality of some sequence data by counting the number of reads mapped to the reference genome and falling within a reference annotation.</li>
			<li><b>match</b>; to ensure good match between the sequence and genotype data. This is useful to detect sample mislabeling, contamination or PCR amplification biases.</li>
			<li><b>quant</b>; to quantify exon and gene expression levels given a reference annotation such as <a hreh="http://www.gencodegenes.org/">GENCODE</a>.</li>
			<li><b>pca</b>; to perform Principal Component Analysis (PCA) on genotype or molecular phenotype data.</li>
			<li><b>cis</b>; to map molecular Quantitative Trait Loci (mQTLs) in <i>cis</i> (i.e. proximal to the phenotype). This mode extends <a href="http://fastqtl.sourceforge.net/">FastQTL</a> by including two additional options: (i) to study groups of phenotypes such as genes containing multiple exons or (ii) to perform conditional analysis in order to discover multiple mQTLs per phenotype</li>
			<li><b>trans</b>; to map QTLs in <i>trans</i>. This can be done either using a full permutation scheme or with a much quicker approximation.</li>
			<li><b>fdensity</b>; to measure the density of functional annotations around the QTL locations. Useful to look at the degree of overlap between QTLs and functional annotations.</li>
			<li><b>fenrich</b>; to measure how QTLs are enriched in any given functional annotations.</li>
			<li><b>rtc</b>; to co-localize collections of QTLs with collection of GWAS hits.</li>
		</ul>
		</p>

		<p>
		<b>QTLtools</b> has been implemented in C++ and has some nice properties:
		<ul>
			<li>It is <b>fast</b>; a lot of efforts have been made to make most of the analyses fast to perform. It notably includes a permutation scheme to map QTLs in <i>cis</i> and in <i>trans</i> that shows very good performance compared to state of the arts methods.</li>
			<li>It is <b>User-friendly</b>; all functional modes have been designed to minimize the work needed to set them up. Usually, a single command line does the trick.</li>
			<li>It is <b>Cluster-friendly</b>; all functional modes have been designed to be easily parallelized on a compute cluster.</li>
		</ul>
		</p>
		
		<p><b>QTLtools</b> is mostly developed by :</p>
		<ul>
			<li>Dr Olivier DELANEAU, University of Geneva, olivier.delaneau[at]unige.ch</li>
			<li>Dr Halit ONGEN, University of Geneva, halit.ongen[at]unige.ch</li>
			<li>Prof. Manolis DERMITZAKIS, university of Geneva, Emmanouil.Dermitzakis[at]unige.ch</li>
		</ul>

		<p>If you use <b>QTLtools</b> in your study, please cite this paper:<br>
		<ul>
			<li> Ongen H, Buil A, Brown AA, Dermitzakis ET, Delaneau O. Fast and efficient QTL mapper for thousands of molecular phenotypes. Bioinformatics. 2016 May 15;32(10):1479-85.</li>
		</ul>
		</p>		
	</div>
	
	<div class="log">Monday 11th July 2016</div>
</div>
</body>
</html>
