<html>
<head>
<meta http-equiv="content-type" content="text/html; charset=iso-8859-1"/>
<meta http-equiv="Cache-Control" content="max-age=3600"/>
<meta name="description" content="description"/>
<meta name="keywords" content="keywords"/> 
<meta name="author" content="author"/> 
<link rel="stylesheet" type="text/css" href="../style/default.css" media="screen"/>
<title>QTLtools</title>
<script language="Javascript" src="../script/print_last_modif_date.js" ></script>
</head>

<body>
<div class="content">

	<div class="item"  id="item1">
		<h1>How to run an approximate pass in trans?</h1>
		<p>
		To perform an approximate analysis in <i>trans</i>, you need to proceed in two steps: 1. build a null distribution and 2. use it to adjust nominal P-values.
		Conversely to the full permutation pass (<a href="mode_trans_full.html" target='content'>link</a>), the null distribution here is designed to adjust nominal P-values for the number of variants being tested. To illustrate this process, first download this example data set:
		</p>

		<ul>
			<li> A phenotype data matrix for 358 samples on chr22: <a href="http://jungle.unige.ch/QTLtools_examples/genes.50percent.chr22.bed.gz" download>BED</a> / <a href="http://jungle.unige.ch/QTLtools_examples/genes.50percent.chr22.bed.gz.tbi" download>index</a></li>
			<li> A genotype data matrix for 358 samples on chr22: <a href="http://jungle.unige.ch/QTLtools_examples/genotypes.chr22.vcf.gz" download>VCF</a> / <a href="http://jungle.unige.ch/QTLtools_examples/genotypes.chr22.vcf.gz.tbi" download>index</a></li>
		</ul>

		<br>
		<h3>Step1: Build the null distribution</h3>
		<p>
		To do so, run the command:
		</p>

		<code>
		QTLtools <b>trans</b> --vcf genotypes.chr22.vcf.gz --bed genes.simulated.chr22.bed.gz <b>--sample 1000 --normal</b> --out trans.sample
		</code>

		<p>
		This basically choose randomly 1,000 phenotypes, permute them and test for association with all variants.
		</p>  

		<note>
		We strongly recommend to use the option <i>--normal</i> here to make sure that all phenotypes are distributed the same way.
		</note>

		<p>
		This produces 3 output files, but you're are interested only in the file <i>trans.null.best.txt.gz</i> that list the best associations discovered for each shuffled phenotype.
		</p>

		<code>
		zcat trans.sample.best.txt.gz | head<br>
		ENSG00000223704.1 -1 4.7752e-06<br>
		ENSG00000025770.14 -1 0.00029513<br>
		ENSG00000183569.13 -1 0.00017113<br>
		ENSG00000226471.2 -1 2.28579e-05<br>
		ENSG00000167077.8 -1 1.84116e-05<br>
		ENSG00000226471.2 -1 3.45359e-05<br>
		ENSG00000211644.2 -1 2.24922e-05<br>
		ENSG00000268292.1 -1 3.26526e-05<br>
		ENSG00000182841.8 -1 6.16083e-05<br>
		ENSG00000241360.1 -1 2.49368e-05
		</code>

		<p>
		The columns in this file are:
		</p>

		<ul>
			<li>1. The ID of the randomly chosen phenotype with sghuffled quantifications</li>
			<li>2. Dummy field, not used here</li>
			<li>3. The smallest nominal P-value discovered for the phenotype</li>
		</ul>

		<p>
		Since <i>--sample 1000</i> has been run, there are 1,000 lines in this file.
		This forms a null distribution that can be used to correct P-values in a nominal pass; similarly to what is done in <i>cis</i>. 
		</p>

		<br>
		<h3>Step2: Adjust the nominal P-values</h3>
		<p>
		To run a nominal pass and adjust the P-values given the null distribution, use:
		</p>

		<code>
		QTLtools <b>trans</b> --vcf genotypes.chr22.vcf.gz --bed genes.50percent.chr22.bed.gz <b>--adjust trans.sample	.best.txt.gz --normal --threshold 0.1</b> --out trans.adjust
		</code>

		<p>
		This does the same than <i>--nominal</i>, excepted that each nominal is adjusted. Specifically, the most notable options used in this command are:
		</p>

		<ul>
			<li><i>--adjust trans.sample.best.txt.gz</i>: to read the null and fit a beta distribution on it</li>
			<li><i>--normal</i>: again, do not forget to add this option in order to be consistent with the step1</li>
			<li><i>--threshold 0.1</i>: this needs to be increased since it now applies on adjusted P-values and not nominals</li>
		</ul>

		<p>
		The output is now corrected for the multiple variants being tested, but we still need to correct for the multiple phenotypes being tested.
		To do so, we proceed in a very similar way than in <i>cis</i>.
		<p>

		<br>
		<h3>Extract significant hits</h3>
		<p>
		Here, We need to determine the adjusted P-value threshold corresponding to a given FDR and then use it to extract all significant hits.
		At this point, we therefore need to look at two particular files that the last command has produced: <i>trans.adjust.best.txt.gz</i> and <i>trans.adjust.hits.txt.gz</i>.
		The first will be used to determine the adjusted P-value threshold that will help to filter out unsignificant hits from the second.
		To run all this, we provide a minimalistic R script that can obviously be adapted, extended and optimized according to your needs.
		To run this script, use:
		</p>

		<code>
		Rscript ./script/runFDR_atrans.R trans.adjust.best.txt.gz trans.adjust.hits.txt.gz 0.05 output.txt
		</code>

		<p>
		The file <i>output.txt</i> is a subset of <i>trans.adjust.hits.txt.gz</i> that basically contains all significant hits at the specified FDR level.
		</p>
	</div>
	
	<div class="log">Monday 11th July 2016</div>
</div>
</body>
</html>
